### PR TITLE
Separate Auth API for Client and Refresher

### PR DESCRIPTION
**Pull request by pankrator**

### CODE REVIEW

File: pkg/auth/oidc/oidc.go [hunk 1]
```diff
@@ -86,69 +89,34 @@ func (s *OpenIDStrategy) Authenticate(user, password string) (*auth.Token, error
...
-	token := &oauth2.Token{
-		AccessToken:  reuseToken.AccessToken,
-		RefreshToken: reuseToken.RefreshToken,
-		Expiry:       reuseToken.ExpiresIn,
-		TokenType:    reuseToken.TokenType,
-	}
-
-	return r.Config.Client(ctx, token)
+func (s *OpenIDStrategy) Client() *http.Client {
+	ctx := context.WithValue(context.Background(), oauth2.HTTPClient, s.httpClient)
```
Comment (pankrator) [inline pkg/auth/oidc/oidc.go]: Should add check if token exists here

File: pkg/auth/oidc/oidc.go [hunk 2]
```diff
@@ -86,69 +89,34 @@ func (s *OpenIDStrategy) Authenticate(user, password string) (*auth.Token, error
...
-// NewTokenRefresher returns new oidc token refresher
-func NewTokenRefresher(config *smclient.ClientConfig, httpClient *http.Client) (auth.TokenRefresher, error) {
-	if config.AuthorizationEndpoint == "" || config.TokenEndpoint == "" {
-		openIDConfig, err := fetchOpenidConfiguration(config.IssuerURL, httpClient.Do)
-		if err != nil {
-			return nil, fmt.Errorf("Error occurred while fetching openid configuration: %s", err)
-		}
-		config.AuthorizationEndpoint = openIDConfig.AuthorizationEndpoint
-		config.TokenEndpoint = openIDConfig.TokenEndpoint
+// Token returns access token, if it expires will refresh it and return it
```
Comment (pankrator) [inline pkg/auth/oidc/oidc.go]: .... if it has expired...

**Comment by pankrator** [comment]
Fixes https://github.com/Peripli/service-manager-cli/issues/20

File: pkg/auth/strategy.go [hunk 3]
```diff
@@ -21,12 +21,17 @@ import (
 	"time"
 )
 
+// Options is used to configure new authenticators and clients
 type Options struct {
 	ClientID              string
```
Comment (KirilKabakchiev) [inline pkg/auth/strategy.go]: add  mapstructure tags as below:

ClientID              string `mapstructure:"client_id"`
	ClientSecret          string `mapstructure:"client_secret"`
	AuthorizationEndpoint string `mapstructure:"authorization_endpoint"`
	TokenEndpoint         string `mapstructure:"token_endpoint"`
	IssuerURL             string `mapstructure:"issuer_url"`

	SSLDisabled bool `json:"ssl_disabled"`

	Timeout time.Duration `json:"timeout"`

**Pull request merged**